File: javascript/packages/flyql/snippets/recipes/error-not-warning-regex.mjs
// Filter messages that match (case-insensitive) 'error' but not 'warning'.
//
// FlyQL query: message ~ "(?i)error" and message !~ "(?i)warning"
//
// The `~` operator is regex match; `!~` is regex non-match. The `(?i)` inline
// flag enables case-insensitive matching.

import { parse } from 'flyql'
import { generateWhere, newColumn } from 'flyql/generators/clickhouse'

const result = parse('message ~ "(?i)error" and message !~ "(?i)warning"')

const columns = {
    message: newColumn({ name: 'message', type: 'String' }),
}

const sql = generateWhere(result.root, columns)
console.log(sql)

File: python/snippets/recipes/error_not_warning_regex.py
"""Filter messages that match (case-insensitive) 'error' but not 'warning'.

FlyQL query: ``message ~ "(?i)error" and message !~ "(?i)warning"``

The ``~`` operator is regex match; ``!~`` is regex non-match. The ``(?i)`` inline
flag enables case-insensitive matching.
"""

from flyql import parse
from flyql.generators.clickhouse.column import Column
from flyql.generators.clickhouse.generator import to_sql_where

result = parse('message ~ "(?i)error" and message !~ "(?i)warning"')

columns = {
    "message": Column("message", "String"),
}

sql = to_sql_where(result.root, columns)
print(sql)
